nuxeo-core/nuxeo-core-test/src/main/java/org/nuxeo/ecm/core/test/FakeSmtpMailServerFeature.java

```diff
@@ -46,44 +77,347 @@
+            }
+
+            return new MailMessage(sm.getHeaderValue("Message-ID"), //
+                    Arrays.asList(sm.getHeaderValues("From")), //
+                    Arrays.asList(sm.getHeaderValues("To")), //
+                    sm.getHeaderValue("Subject"), //
+                    sm.getHeaderValue("Content-Type"), //
+                    sm.getBody(), date);
+        }
+    }
+
+    /**
+     * Adapter of the {@link SmtpMessage}.
+     * 
+     * @since 11.1
+     */
+    public static class MailMessage {
+
+        protected String id;
+
+        protected List<String> senders;
+
+        protected List<String> recipients;
+
+        protected String subject;
+
+        protected String content;
+
+        protected ZonedDateTime date;
+
+        protected String contentType;
+
+        public MailMessage(String id, List<String> senders, List<String> recipients, String subject, String contentType,
+                String content, ZonedDateTime date) {
+            this.id = requireNonNull(id);
+            this.senders = requireNonNullElseGet(senders, List::of);
+            this.recipients = requireNonNullElseGet(recipients, List::of);
+            this.subject = subject;
+            this.contentType = contentType;
+            this.content = content;
+            this.date = date;
+        }
+
+        public String getId() {
+            return id;
+        }
+
+        public List<String> getSenders() {
+            return senders;
+        }
+
+        public List<String> getRecipients() {
+            return recipients;
+        }
+
+        public String getSubject() {
+            return subject;
+        }
+
+        public String getContent() {
+            return content;
+        }
+
+        public ZonedDateTime getDate() {
+            return date;
+        }
+
+        public String getContentType() {
+            return contentType;
+        }
+
+        public String getText() {
+            if (getContentType().contains("text/html")) {
+                Document document = Jsoup.parse(getContent());
+                return document.select("body").text();
+            }
+            return getContent();
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+
+            if (o == null || getClass() != o.getClass())
+                return false;
+
+            MailMessage that = (MailMessage) o;
+
+            return new EqualsBuilder().append(id, that.id).isEquals();
+        }
+
+        @Override
+        public int hashCode() {
+            return new HashCodeBuilder(17, 37).append(id).toHashCode();
         }
     }
+
+    /**
+     * The {@link MailCapture} annotation tells {@link FakeSmtpMailServerFeature} that the method to which it is
+     * attached is interesting to capture the mails that will be sent. In other hand the {@link MailCapture} is a marker
+     * for the tests methods that will focus on checking the sent mails.
```

Comment:
This needs to be reworded, I propose this:
```
     * The {@link MailCapture} annotation tells the {@link FakeSmtpMailServerFeature} the method it is attached to
     * is interested in capturing the mails that will be sent. The {@link MailCapture} is also a marker
     * for the tests methods focused on checking the sent mails.
```

Reply:
this annotation will be removed waiting for [https://github.com/nuxeo/nuxeo/pull/3599/](PR) on the master